### PR TITLE
Run sheet to have digits in the file name

### PR DESCRIPTION
**Pull request by kcho**
updates
- feat: run sheet to have digits in the file name
- fix: mediaflux to keep zoom folder structure
- skip pulling files when there is error in the RPMS exported CSV files.

### CODE REVIEW

**Comment by kcho** [comment]
Related issue https://github.com/AMP-SCZ/lochness/issues/102

File: lochness/rpms/__init__.py [hunk 1]
```diff
@@ -235,20 +250,26 @@ def get_subject_data(all_df_dict: Dict[str, pd.DataFrame],
...
-            else:
-                most_recent_row_index = pd.to_datetime(
-                        table['LastModifiedDate']).idxmax()
-                non_recent_row_index = [x for x in table.index
-                         if x != most_recent_row_index]
-                print(f'RPMS export has duplicated rows for {measure}')
-                subject_df.drop(non_recent_row_index, inplace=True)
+        # RPMS should overwrite the row whenever there is an update in any 
+        # field of the visit. The snippet below is a safety measure, to store
+        # most recent visit row for each visit
```
Comment (tashrifbillah) [inline lochness/rpms/__init__.py:261]: A continuation of the thought I shared on MS Teams, do RPMS keep all rows of all visits until that time?

File: lochness/rpms/__init__.py [hunk 2]
```diff
@@ -188,9 +202,10 @@ def initialize_metadata(Lochness: 'Lochness object',
                                         df_measure[rpms_id_colname]
 
             # if mindlamp_id exists in the rpms table
-            if 'mindlamp_id' in df_measure:
-                subject_dict['mindlamp'] = f'mindlamp.{study_name}:' \
-                        + df_measure[f'mindlamp_id']
+            if 'chrdbb_lamp_id' in df_measure:
+                if not pd.isna(df_measure[f'chrdbb_lamp_id']):
+                    subject_dict['Mindlamp'] = f'mindlamp.{study_name}:' \
+                            + df_measure[f'chrdbb_lamp_id']
```
Comment (tashrifbillah) [inline lochness/rpms/__init__.py:208]: Can you confirm that this block is unrelated to the feature I requested?

File: lochness/rpms/__init__.py [hunk 3]
```diff
@@ -167,6 +174,13 @@ def initialize_metadata(Lochness: 'Lochness object',
         for index, df_measure in df_measure_all_subj.iterrows():
             # if multistudy:
             # site of the subject for the line
+
+            # if the rpms table is not ready (e.g.doesn't have the subject col)
+            if rpms_id_colname not in df_measure.index or \
+                pd.isna(df_measure[rpms_id_colname]):
+                continue
```
Comment (tashrifbillah) [inline lochness/rpms/__init__.py:181]: Just saying out loud--this block will be ineffectual once all forms are final.

File: lochness/rpms/__init__.py [hunk 4]
```diff
@@ -110,22 +110,29 @@ def get_run_sheets_for_datatypes(target_df_loc: Union[Path, str]) -> None:
...
+
+            # convert month to datapoint
+            if modality in ['eeg', 'mri']:
+                # two month : first timepoint
+                # four month : second timpeoint
+                time_to_timepoint = {2: 1, 4: 2}
+            elif fieldname == 'PennCNB':
+                time_to_timepoint = {2: 1, 4: 2, 8: 3, 14: 4, 16: 5}
+            else:  # actigraphy & EMA
+                time_to_timepoint = dict(zip(range(2, 15), range(1, 14)))
```
Comment (tashrifbillah) [inline lochness/rpms/__init__.py:127]: There are many other forms that are assigned to multiple time points:
image

Can you confirm that the comment `# actigraphy & EMA` is underinclusive?

**Review by tashrifbillah — COMMENTED** [review]
Changes in `mediaflux/__init__.py` and `redcap/__init__.py` are unrelated to feature request #102. Main changes in `rpms/__init__.py` look good to Tashrif. Yet, he has asked a few clarifying questions.

File: lochness/rpms/__init__.py (hunk 2 above)
Comment (kcho) [reply lochness/rpms/__init__.py:208]: Correct. This is resolving `mindlamp` run sheet name change issue.

File: lochness/rpms/__init__.py (hunk 4 above)
Comment (kcho) [reply lochness/rpms/__init__.py:127]: It is according to the table that Larry shared in the email. We may have to change this in future PR, but it works fine for now.
